apps/framework-docs/src/pages/moose/olap/db-pull.mdx

```diff
@@ -0,0 +1,143 @@
+---
+title: Syncing External Tables
+description: Refresh your external table models from an existing ClickHouse database
+---
+
+import { Callout, LanguageSwitcher, TypeScript, Python, BulletPointsCard } from "@/components";
+
+# Syncing External Tables
+
+<LanguageSwitcher />
+
+## What this is
+
+Use `moose db pull` to refresh the definitions of tables you marked as `EXTERNALLY_MANAGED` from a live ClickHouse instance. It reads your code to find external tables, fetches their remote schemas, regenerates one external models file, and creates a small git commit if anything changed.
+
+## When to use it
+
+- **External tables changed remotely**: a DBA, CDC, or ETL pipeline updated schema.
+- **Keep types in sync**: update generated models without touching fully-managed tables.
+- **Safe by design**: does not modify the database or your managed models.
+
+<Callout type="info">This is a read-only sync for your code models. For concepts and modeling guidance, see [External Tables](/moose/olap/external-tables). To bootstrap a project from an existing DB, see [Initialize from ClickHouse](/moose/getting-started/from-clickhouse).</Callout>
+
+## Requirements
+
+- Tables are defined with <TypeScript inline>`lifeCycle: EXTERNALLY_MANAGED`</TypeScript><Python inline>`life_cycle=LifeCycle.EXTERNALLY_MANAGED`</Python>
+- A ClickHouse connection string (native or HTTP/S)
+
+## Connection strings
+
+`db pull` accepts both native and HTTP(S) URLs. Native strings are automatically converted to HTTP(S) with the appropriate ports.
+
+Examples:
+
+```bash filename="Terminal" copy
+# Native (auto-converted to HTTPS + 8443)
+moose db pull --connection-string "clickhouse://explorer@play.clickhouse.com:9440/default"
+
+# HTTPS (explicit database via query param)
+moose db pull --connection-string "https://play.clickhouse.com/?user=explorer&database=default"
+
+# Local HTTP
+moose db pull --connection-string "http://localhost:8123/?user=default&database=default"
+```
+
+## What gets written
+
+- TypeScript: `app/externalModels.ts`
+- Python: `app/external_models.py`
+
+The file contains only the tables you marked as `EXTERNALLY_MANAGED` and starts with an auto-generated header.
```

Comment:
"The file contains only the tables you marked as `EXTERNALLY_MANAGED`" -- this is kind of confusing? is this saying that it will copy any OLAPTable definitions from anywhere in my code base into this file (if they are marked externally managed)? won't i have duplicate declarations then? or does it only update table definitions in this file?

I would expect it to update all my externally managed table definitions in place, wherever they are in my code base. or we need to be clearer that the externalModels file is not just a recommended convention but a required structure that all externally managed tables have to be there. (at which point why have the config setting on the object.. just base it on the file)

Reply:
hm, ok, after reading further, i'm interpreting it to be "it will copy any OLAPTable definitions from anywhere in my code base into this file (if they are marked externally managed)". which seems a little odd, but ok